### PR TITLE
Updated to support Express 4.x

### PR DESCRIPTION
**Pull request by kmchugh**
Express 4.x no longer represents res.locals as a function, this fix allows the treatment of res.locals as an object for 4.x and keeps backwards compatibility in 3.x and 2.x.

This also fixes up race conditions in express-test.js

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@kmchugh - Thanks for the PR Ken! I am sorry that it has taken so long to get it looked at. I have done an initial code review and things are looking great. I have not tested yet, but the one blocker I see for merge is spacing. Can you go back to the original formatting and use 2 space tabs?

File: lib/connect-cachify.js [hunk 1]
```diff
@@ -17,124 +17,134 @@ var existsSync = fs.existsSync || path.existsSync,
...
-            _cache[true_path] = {exists: true};
-          } else {
-            // Hmm potential DOS Attack here... maybe we shouldn't cache
-            // checking disk every time isn't that bad - just like static
-            // middleware
-            // Or use an LRU?
-            console.warn('Cachify like URL, but no file found');
-            _cache[true_path] = {exists: false};
-          }
+            resp.locals.cachify_prefetch = cachify_prefetch,
```
Comment (shane-tomlinson) [inline lib/connect-cachify.js:73]: use `;` to terminate a line instead of `,`.

File: lib/connect-cachify.js [hunk 2]
```diff
@@ -17,124 +17,134 @@ var existsSync = fs.existsSync || path.existsSync,
...
+            resp.locals.cachify_js = cachify_js,
+            resp.locals.cachify_css = cachify_css,
+            resp.locals.cachify = cachify
```
Comment (shane-tomlinson) [inline lib/connect-cachify.js:76]: missing `;`

**Comment by shane-tomlinson** [comment]
@kmchugh - I am closing this and opening a new PR with my requests. Thank you very much!

**Comment by kmchugh** [comment]
Thanks Shane,
Much appreciated!

**Pull request closed without merging**